daemon/response.go

```diff
@@ -473,6 +473,24 @@ func SnapChangeConflict(cce *snapstate.ChangeConflictError) Response {
 	}
 }
 
+// InsufficientSpace is an error responder used when an operation cannot
+// be performed due to low disk space.
+func InsufficientSpace(dserr *snapstate.InsufficientSpaceError) Response {
+	value := map[string]interface{}{
+		"snap-names": dserr.Snaps,
+		"change-kind": dserr.ChangeKind,
+	}
+	return &resp{
+		Type: ResponseTypeError,
+		Result: &errorResult{
+			Message: dserr.Error(),
+			Kind: client.ErrorKindInsufficientSpace,
+			Value: value,
+		},
+		Status: 409,
```

Comment:
I suppose that should be 412?

Reply:
no, that one has a very precise meaning.  There is no great code for our case. See the discussion here for example:

https://evertpot.com/http/507-insufficient-storage

Reply:
Ok, sounds like 507 it is then for the lack of better candidate? 413 mentioned there doesn't really fit.

Reply:
Given that none really matches, should we just make up a 4XX error code?

Reply:
we'll have to document all of this for 2.48, and things wil be behind a feature flag for a bit, let's go with 507 for now and see what feedback we get